sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/AsyncCache.java

```diff
@@ -127,7 +128,9 @@ public void refresh(
         logger.debug("refreshing cache[{}]", key);
         AsyncLazy<TValue> initialLazyValue = values.get(key);
         if (initialLazyValue != null && (initialLazyValue.isSucceeded() || initialLazyValue.isFaulted())) {
-            AsyncLazy<TValue> newLazyValue = new AsyncLazy<>(singleValueInitFunc);
+            // TODO: on the force refresh we are keeping the old value just in case the current refresh fails.
+            // is this the correct behaviour? or should we forget the old value as it is a "forced" refresh?
+            AsyncLazy<TValue> newLazyValue = new AsyncLazy<>(singleValueInitFunc, initialLazyValue.getOldValue());
```

Comment:
We should probably got through the different caches - My initial reaction is that for address refresh we should always hold on to the new value  - for the PartitionKeyRangeCache and CollectionCache we should probably rather not hold-onto the old value